### PR TITLE
Db enum inconsistency

### PR DESCRIPTION
**Pull request by sanderr**
# Description

Fixed inconsistency and backwards compatibility for deprecated processing_events resource state (introduced in #3020). We previously deleted the `processing_events` resource state because it was no longer required. We didn't delete it from the database however. This was a concious decision at the time to stay backwards compatible with older agents, but it turns out we overlooked some details: neither the API, nor the data model allowed the deprecated value anymore, causing any such values remaining in the database to result in errors.

This pull request removes the value from the database enum, and adds an explicit conversion to the `deploying` state in the entrypoint that might be called by older agents, for backwards compatibility.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/test_data.py [hunk 1]
```diff
@@ -102,6 +105,34 @@ async def test_postgres_client(postgresql_client):
...
+                INNER JOIN pg_type ON pg_enum.enumtypid = pg_type.oid
+                INNER JOIN information_schema.columns c ON pg_type.typname = c.udt_name
+                WHERE table_schema='public' AND table_name=$1 AND column_name=$2
+                """,
+                cls._get_value(cls.table_name()),
+                cls._get_value(enum_column),
+            )
+            for record in db_enum_values:
+                # verify that all db-defined values can be parsed by the field definition object
+                assert isinstance(field._from_db_single(enum_column, record["enumlabel"]), enum.Enum)
```
Comment (wouterdb) [inline tests/test_data.py:133]: perhaps we have to also assert that all python enum values are also allowed in the DB?

**Review by wouterdb — APPROVED** [review]

File: tests/test_data.py (hunk 1 above)
Comment (sanderr) [reply tests/test_data.py:133]: Yes, good catch, I'll add that.

**Review by Hugo-Inmanta — COMMENTED** [review]
Partial review so far, I'll continue tomorrow with a fresh pair of eyes.

File: tests/test_data.py (hunk 1 above)
Comment (sanderr) [reply tests/test_data.py]: Turns out this wasn't the case. Re-requesting review.

**Review by wouterdb — APPROVED** [review]

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]
I'm not super familiar with db migrations so I understand the intent of the tests in tests/db/test_v202208180_to_v202209130.py and tests/test_data.py but I must admit the execution is still pretty cryptic to me.

File: src/inmanta/db/versions/v202209130.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        "skipped",
+        "dry",
+        "deployed",
+        "failed",
+        "deploying",
+        "available",
+        "cancelled",
+        "undefined",
+        "skipped_for_undefined",
+    ]
```
[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/db/versions/v202209130.py:42]: Exactly!

File: tests/db/test_v202208180_to_v202209130.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+def mock_enum(monkeypatch) -> None:
+    enum_meta: Type[Type[Enum]] = type(Enum)
+    old_instancecheck: abc.Callable[[Type[Type[Enum]], object], bool] = enum_meta.__instancecheck__
+    monkeypatch.setattr(
+        # accept MockEnum anywhere any specific enum is expected
+        enum_meta,
+        "__instancecheck__",
+        lambda cls, instance: old_instancecheck(cls, instance) or isinstance(instance, MockEnum),
+    )
```
[the remark this answers is not in the document]
Comment (sanderr) [reply tests/db/test_v202208180_to_v202209130.py:47]: Nasty stuff (though interesting), I'm not proud of it, but in test cases I allow myself the occasional hack.

File: src/inmanta/const.py [hunk 4]
```diff
@@ -34,6 +34,34 @@ class ResourceState(str, Enum):
...
+class NonDeployingResourceState(str, Enum):
+    unavailable = ResourceState.unavailable.value
+    skipped = ResourceState.skipped.value
+    dry = ResourceState.dry.value
+    deployed = ResourceState.deployed.value
+    failed = ResourceState.failed.value
+    available = ResourceState.available.value
+    cancelled = ResourceState.cancelled.value
+    undefined = ResourceState.undefined.value
+    skipped_for_undefined = ResourceState.skipped_for_undefined.value
```
[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/const.py:46]: So I did. I actually went back to this implementation for better mypy support but forgot to push the changes. So I kept this one, deleted the other one.

**Review by Hugo-Inmanta — APPROVED** [review]

**Review by wouterdb — APPROVED** [review]

**Pull request closed without merging**